x/supplier/keeper/proof.go

```diff
@@ -7,25 +7,27 @@ import (
 	"github.com/pokt-network/poktroll/x/supplier/types"
 )
 
-// SetProof set a specific proof in the store from its index
-func (k Keeper) SetProof(ctx sdk.Context, proof types.Proof) {
+// UpsertProof inserts or updates a specific proof in the store by index.
```

Comment:
What's the purpose of updating a claim? I'm missing context on when this should be done/

Reply:
👍🏼 To this I am unsure why they are updateable?

Reply:
These are good points and took me a little while to think through.  See my response in `x/supplier/keeper/claim.go`.

Hypothetically, anyone could write their own client to post claims/proofs to the on-chain logic. If they made a mistake (e.g. malformed request), we shouldn't penalize the actor if the submission window hasn't closed. 

@bryanchriswhite Can you please add similar comments to the proof msg server (window & proof validation)?